### PR TITLE
Read & Query HTTP APIs - Part 2

### PR DESCRIPTION
**Pull request by csuwildcat**

### CODE REVIEW

File: src/http-api.ts [hunk 1]
```diff
@@ -104,28 +129,83 @@ export class HttpApi {
...
+          });
+          const reply = await this.dwn.processMessage(req.params.did, record.toJSON());
+          return readReplyHandler(res, reply);
+        }
+        else {
+          return res.sendStatus(404);
+        }
+      }
+      else if (status.code === 401) {
+        return res.sendStatus(404);
```
Comment (LiranCohen) [inline src/http-api.ts:157]: Sanity: same question as above re 404 vs 401

File: src/http-api.ts [hunk 2]
```diff
@@ -104,28 +129,83 @@ export class HttpApi {
...
+      }
+      else {
+        return res.sendStatus(status.code);
+      }
+    })
+
+    this.#api.get('/:did/read/protocols/:protocol', async (req, res) => {
```
Comment (LiranCohen) [inline src/http-api.ts:164]: Nit: Write a short comment above each of the endpoints describing it's intended behavior.

At some point in a different effort we could expand on that and find a suitable tool to generate API Endpoint documentation.

**Review by LiranCohen — COMMENTED** [review]
Looks good, just a question regarding returning 404 instead of 401.

File: src/http-api.ts (hunk 2 above)
Comment (thehenrytsai) [reply src/http-api.ts:164]: Alternatively better still (or in addition), just add a `HTTP API` section next to JSON-RPC API section to quickly describe these newly added API (including `query`), dev-rel folks and everyone else will need to know this info anyway.

**Review by thehenrytsai — COMMENTED** [review]
Looks good, made a bunch of refactoring. Only tests is left to bring new code coverage up%.

**Comment by csuwildcat** [comment]
Let's just merge it

**Review by LiranCohen — APPROVED** [review]
After adding some tests this is in a good place.

Once https://github.com/TBD54566975/dwn-sdk-js/issues/739 is complete, we can clean up some of this code that does a `RecordsQuery` followed by a `RecordsRead` with only the `RecordsRead`.

This will also allow us to clean up some of the potentially unreachable codepaths in place today.

**Review by thehenrytsai — APPROVED** [review]
A send-it approval, given the amount of tests that were added.

🐐 @LiranCohen.

**Pull request merged**